### PR TITLE
FC-783 nodejs improvements/changes for closed api

### PR DESCRIPTION
**Pull request by ldw1007**
summary of changes by file/class:

flureenjs
* bumped version

fluree.db.api.ledger
* retrieve a permissioned ledger for nodejs (vs root) based on the provided authentication information (e.g., JWT, private key/auth)

fluree.db.connection
* corrected typo
* removed unused headers from default-storage-read
* modified generate-connection to clear db cache for nodejs

fluree.db.connection_js
* marked function(s) as deprecated - can remove after other javascript libraries are updated

fluree.db.dbfunctions.core
* extracted/consolidates find-fn logic for clj/nodejs usage

fluree.db.query.block
* re-enable permission checks for nodejs 
* extracted/consolidated filter logic for clj/nodejs

fluree.db.query.range
* re-enabled permission checks for nodejs

### CODE REVIEW

File: src/fluree/db/connection.cljc [hunk 1]
```diff
@@ -434,6 +432,25 @@
...
 
 
+
+(defn add-token
+  "Adds token to connection information so it is available to submit storage read requests.
+
+  Returns true if successful, false otherwise."
+  [conn token]
+  (let [conn-id (:id conn)
+        _ (log/info "add-token" {:conn-id conn-id :token token})
```
Comment (bplatz) [inline src/fluree/db/connection.cljc:442]: Debugging left in?

File: src/fluree/db/connection.cljc [hunk 2]
```diff
@@ -434,6 +432,25 @@
...
+        ]
+    (try
+      (swap! server-connections-atom update-in [conn-id :token]
```
Comment (bplatz) [inline src/fluree/db/connection.cljc:445]: Can use: (swap! server-connections-atom update-in [conn-id :token] #(or % token))

Alternatively if there could a boolean "false" value the above would overwrite it, but alternative if that is a concern could be:
(swap! server-connections-atom update-in [conn-id :token] (fnil identity token))

File: src/fluree/db/connection.cljc [hunk 3]
```diff
@@ -481,8 +498,10 @@
                              (async/close! pub-chan)
                              (close-websocket conn-id)
                              (swap! state-atom assoc :close? true)
-                             ;; NOTE - when we allow permissions back in CLJS, remove conditional below
-                             #?(:clj (dbfunctions/clear-db-fn-cache))
+                             ;; NOTE - when we allow permissions back in CLJS (browser), remove conditional below
+                             #?(:clj (dbfunctions/clear-db-fn-cache)
+                                :cljs (if (identical? "nodejs" cljs.core/*target*)
```
Comment (bplatz) [inline src/fluree/db/connection.cljc:503]: Use a 'when'  in this case, not 'if' (there is no else condition)

File: src/fluree/db/connection.cljc [hunk 4]
```diff
@@ -434,6 +432,25 @@
...
+        _ (log/info "add-token" {:conn-id conn-id :token token})
+        ]
+    (try
+      (swap! server-connections-atom update-in [conn-id :token]
+             (fn [x]
+               (if x
+                 x
+                 token)))
+      true
+      (catch #?(:clj Exception :cljs :default) _
```
Comment (bplatz) [inline src/fluree/db/connection.cljc:451]: Can use our new try*/catch* macro that automatically resolves the :clj/:cljs conditions so you don't need to do that here.

File: src/fluree/db/query/block.cljc [hunk 5]
```diff
@@ -1,14 +1,22 @@
...
-;; TODO - exists for nodejs compilation
+(defn- filter-block-flakes
+  "Applies filter(s) to flakes in a block"
+  [db block]
+  (let [root? (-> db :permissions :root?)]
+    (if root?
+      block
+      {:block  (:block block)
+       :t      (:t block)
+       :flakes (<? (perm-validate/allow-flakes? db (:flakes block)))})))
```
Comment (bplatz) [inline src/fluree/db/query/block.cljc:18]: This shouldn't be working - not in a go-block

File: src/fluree/db/query/block.cljc [hunk 6]
```diff
@@ -22,12 +30,10 @@
...
-            res #?(:cljs res                                ;; always allow for now
-                   :clj (if root?
-                          res
-                          {:block  (:block res)
-                           :t      (:t res)
-                           :flakes (<? (perm-validate/allow-flakes? db (:flakes res)))}))
+            res #?(:cljs (if (identical? "nodejs" cljs.core/*target*)
+                           (filter-block-flakes db res)
+                           res)  ;; browser: always allow for now
+                   :clj (filter-block-flakes db res))
```
Comment (bplatz) [inline src/fluree/db/query/block.cljc:36]: This as it is written is async, and would need a <? (after wrapping it in a go-block)

**Review by bplatz — COMMENTED** [review]
The filtering code not in a go-block I would think will fail tests  as it would not end up doing proper permissions... if it isn't failing the test we should look at the test to make sure why.

**Comment by ldw1007** [comment]
The block-range worked for :root/open-api?  I verified the latest changes when a closed-api is used

**Review by bplatz — APPROVED** [review]

**Pull request merged**